upsilonconf/io.py

```diff
@@ -225,26 +225,109 @@ def _get_io_function(path: Path, write: bool = False):
     raise ValueError(f"unknown config file extension: '{ext}'")
 
 
-def load(path: Union[Path, str]) -> Configuration:
+def __replace_in_keys(
+        dictionary: Union[Dict[str, Any], list, Any],
+        s: str,
+        r: str) -> Union[Dict[str, Any], list, Any]:
+    """
+    Take a list or dictionary and replace all occurrencies of `s` in any of its
+    keys with `r`.
+
+    This function is called recursively.
+
+    Modelled after the Stack Overflow answer by Farhan Haider:
+    https://stackoverflow.com/questions/21650850/pyyaml-replace-dash-in-keys-with-underscore#answer-55986782
+
+    Parameters
+    ----------
+    dictionary : dict|list
+        The list or dictionary to be modified.
+    s : str
+        The string to be replaced.
+    r : str
+        The replacement string.
+
+    Returns
+    -------
+        The object of the same type with all strings `s` replaced by `r`.
+    """
+    # For lists perform this method on every object
+    if isinstance(dictionary, list):
```

Comment:
If I don't mistake, there is no support for list-configuration objects. A list of dicts remains a list of dicts, so there should be no need to fix keys in this case (if I don't mistake).

Reply:
Well, if that is so, we can certainly remove it (or comment it out), if you are sure, that this will never be needed.

Should I remove it then?

Reply:
Unless you tried something that fails without this piece of code, I would remove it. This should also obliviate the need for any type-checks (something I personally do not like).

Reply:
Done in https://github.com/hoedt/upsilonconf/pull/3/commits/5f036c7ccf37f9bdccd57ee9802deac4bcb3ce01.